src/cockpit/beiboot.py

```diff
@@ -212,60 +245,68 @@ def __init__(self, router: Router, destination: str, args: argparse.Namespace):
         super().__init__(router)
 
     async def do_connect_transport(self) -> None:
-        # do we have user/password (Basic auth) from the login page?
-        auth = await self.router.request_authorization("*")
-        ssh_opts = []
+        # Choose your own adventure...
+        if os.path.exists('/.flatpak-info'):
+            await self.connect_from_flatpak()
+        else:
+            await self.connect_from_bastion_host()
+
+    async def connect_from_flatpak(self) -> None:
+        # We want to run a python interpreter somewhere...
+        cmd, env = python_interpreter('cockpit-bridge')
```

Comment:
The signature of python_interpreter() and flatpak_spawn() is weird -- it returns an "env" which is always empty. This makes this code hard to read/understand. I.e. this here is nothing more than an initialization of `env = []`, which is easier to understand.

Reply:
The idea is that we have a bunch of these functions and they return a pair of argv/envp.  The next thing in the chain expects an envp (and needs to be able to handle it) so it has to come from somewhere, and this is the most natural place...

Reply:
Yeah, I got it -- I figure we should make all functions *receive* an env as well then, and return the concatenation.

Reply:
Not important, I left it as-is for now.